Automattic/pocket-casts-android
# Mock product details in debug builds when offers are empty

## Description

This PR adds product details mocking to debug builds. I used build variants to make sure that this code never lands in the release builds.

Closes #2624

## Testing Instructions

1. Build `release` variant and sign in with a free account.
2. You should see the real paywall offer.
3. Go to the account page.
4. You should see the real offer cards.
5. Build `debugProd` variant and sign in with a free account.
6. You should see a mocked paywall offer.
7. Got to the account page
8. You should see a mocked offer cards.
9. Build `debug` variant and sign in with a free account.
10. You should see a mocked paywall offer.
11. Got to the account page
12. You should see a mocked offer cards.

## Checklist
 
#### I have tested any UI changes...
